src/charm.py

```diff
@@ -212,22 +229,38 @@ def _on_relation_departed(self, event: RelationDepartedEvent, charm_state: Charm
             event: relation departed event.
             charm_state: The charm state.
         """
+        peer_relation = self.model.relations[synapse.SYNAPSE_PEER_RELATION_NAME]
+        if peer_relation is not None and self.unit.is_leader():
+            peer_units_total = peer_relation[0].data[self.app].get(PEER_UNITS_TOTAL, "1")
+            peer_relation[0].data[self.app].update(
+                {PEER_UNITS_TOTAL: str(int(peer_units_total) - 1)}
+            )
         if event.departing_unit == self.unit:
             # there is no action for the departing unit
             return
-        peer_relation = self.model.get_relation(synapse.SYNAPSE_PEER_RELATION_NAME)
-        if peer_relation is not None:
-            # if more than 1, check for Redis integration
-            synapse_units = len(peer_relation.units)
-            if synapse_units > 1:
-                logger.debug("Found %d units, checking for Redis integration.", synapse_units)
-                if charm_state.redis_config is None:
-                    logger.debug("Scaling is not allowed without Redis integration.")
-                    self.model.unit.status = ops.BlockedStatus(
-                        "Redis integration not found. Please, verify it."
-                    )
-                    return
-        self.model.unit.status = ops.ActiveStatus()
+        if (
+            event.departing_unit
+            and event.departing_unit.name == self.get_main_unit()
+            and self.unit.is_leader()
+        ):
+            # Main is gone so I'm the leader and will be the new main
+            self.set_main_unit(self.unit.name)
+            self.change_config(charm_state)
+
+    def peer_units_total(self) -> int:
+        """Get peer units total.
+
+        Returns:
+            total of units in peer relation or None if there is no peer relation.
+        """
+        peer_relation = self.model.relations[synapse.SYNAPSE_PEER_RELATION_NAME]
+        if not peer_relation:
+            logger.error(
+                "Failed to get main unit: no peer relation %s found",
+                synapse.SYNAPSE_PEER_RELATION_NAME,
+            )
+            return 1
+        return int(peer_relation[0].data[self.app].get(PEER_UNITS_TOTAL, "1"))
```

Comment:
May I ask if there is a specific reason for counting units in the peer relation rather than using `ops.CharmBase.app.planned_units`?

Reply:
I was a little bit concerned about this:
"Pending units are included in the count, and scale down events may
        modify the count before some units have been fully torn down. The information in
        planned_units is up-to-date as of the start of the current hook invocation."

Reply:
While working on the next story I thought..."why not?"
So I changed here to use. I just took some time to found out that for the unit tests I needed to set the units with harness.set_planned_units(2)
Thanks!